.upstream-tests/test/std/numerics/complex.number/cases.h

```diff
@@ -16,166 +16,200 @@
-
-    cuda::std::complex<double>(NAN, -0.),
-    cuda::std::complex<double>(-INFINITY, -0.),
-    cuda::std::complex<double>(-2, -0.),
-    cuda::std::complex<double>(-1, -0.),
-    cuda::std::complex<double>(-0.5, -0.),
-    cuda::std::complex<double>(-0., -0.),
-    cuda::std::complex<double>(+0., -0.),
-    cuda::std::complex<double>(0.5, -0.),
-    cuda::std::complex<double>(1, -0.),
-    cuda::std::complex<double>(2, -0.),
-    cuda::std::complex<double>(INFINITY, -0.),
-
-    cuda::std::complex<double>(NAN, +0.),
-    cuda::std::complex<double>(-INFINITY, +0.),
-    cuda::std::complex<double>(-2, +0.),
-    cuda::std::complex<double>(-1, +0.),
-    cuda::std::complex<double>(-0.5, +0.),
-    cuda::std::complex<double>(-0., +0.),
-    cuda::std::complex<double>(+0., +0.),
-    cuda::std::complex<double>(0.5, +0.),
-    cuda::std::complex<double>(1, +0.),
-    cuda::std::complex<double>(2, +0.),
-    cuda::std::complex<double>(INFINITY, +0.),
-
-    cuda::std::complex<double>(NAN, 0.5),
-    cuda::std::complex<double>(-INFINITY, 0.5),
-    cuda::std::complex<double>(-2, 0.5),
-    cuda::std::complex<double>(-1, 0.5),
-    cuda::std::complex<double>(-0.5, 0.5),
-    cuda::std::complex<double>(-0., 0.5),
-    cuda::std::complex<double>(+0., 0.5),
-    cuda::std::complex<double>(0.5, 0.5),
-    cuda::std::complex<double>(1, 0.5),
-    cuda::std::complex<double>(2, 0.5),
-    cuda::std::complex<double>(INFINITY, 0.5),
-
-    cuda::std::complex<double>(NAN, 1),
-    cuda::std::complex<double>(-INFINITY, 1),
-    cuda::std::complex<double>(-2, 1),
-    cuda::std::complex<double>(-1, 1),
-    cuda::std::complex<double>(-0.5, 1),
-    cuda::std::complex<double>(-0., 1),
-    cuda::std::complex<double>(+0., 1),
-    cuda::std::complex<double>(0.5, 1),
-    cuda::std::complex<double>(1, 1),
-    cuda::std::complex<double>(2, 1),
-    cuda::std::complex<double>(INFINITY, 1),
-
-    cuda::std::complex<double>(NAN, 2),
-    cuda::std::complex<double>(-INFINITY, 2),
-    cuda::std::complex<double>(-2, 2),
-    cuda::std::complex<double>(-1, 2),
-    cuda::std::complex<double>(-0.5, 2),
-    cuda::std::complex<double>(-0., 2),
-    cuda::std::complex<double>(+0., 2),
-    cuda::std::complex<double>(0.5, 2),
-    cuda::std::complex<double>(1, 2),
-    cuda::std::complex<double>(2, 2),
-    cuda::std::complex<double>(INFINITY, 2),
-
-    cuda::std::complex<double>(NAN, INFINITY),
-    cuda::std::complex<double>(-INFINITY, INFINITY),
-    cuda::std::complex<double>(-2, INFINITY),
-    cuda::std::complex<double>(-1, INFINITY),
-    cuda::std::complex<double>(-0.5, INFINITY),
-    cuda::std::complex<double>(-0., INFINITY),
-    cuda::std::complex<double>(+0., INFINITY),
-    cuda::std::complex<double>(0.5, INFINITY),
-    cuda::std::complex<double>(1, INFINITY),
-    cuda::std::complex<double>(2, INFINITY),
-    cuda::std::complex<double>(INFINITY, INFINITY)
+using testcases_t = cuda::std::complex<double>[137];
+
+struct _testcases {
```

Comment:
Does this exist to avoid needing the array overloads of being/end? If so, why is that needed? Do we just not expose them yet?

Reply:
I don't think we expose begin/end yet. Some tests might rely on range based for loops, so I just wanted satisfy that as non-obtrusively as possible.

NVRTC was particularly angry about dynamic initialization so I had to wrap up the array to make it work.

Reply:
Cool, ship it!